api/src/main/java/org/pmiops/workbench/db/dao/DataSetServiceImpl.java

```diff
@@ -358,11 +538,12 @@ private String getQualifiedColumnName(Domain currentDomain, String columnName) {
             "Kernel Type " + kernelTypeEnum.toString() + " not supported");
     }
     return queryJobConfigurationMap.entrySet().stream()
+        .filter(e -> e.getKey() != null)
```

Comment:
This is an interesting addition – when do we expect to see a map entry with a null key? Seems worth a comment, since the reason for this feels unclear from context.

Reply:
Seems like we had that before due to some pre-stream map lookup into list logic that required a null check when going over the list. Good news is ImmutableList won't allow nulls.